### ISSUE TITLE
Are borrowed properties allowed in binding or assignment patterns?

### ISSUE THREAD

**Issue by zenparsing** [RReverser/es-borrowed-props#4]
Should this be a syntax error?

``` js
({ a.b }) = obj;
```

If so then there will probably need to be a cover grammar (similar to how CoverInitializedName works).

**Comment by RReverser** [comment]
No, it shouldn't - that's why I extended `CoverInitializedName` with `BorrowedMemberExpression` in order to make it work as `({ b: a.b } = obj)`

**Comment by RReverser** [comment]
Ah wait did you mean specifically `({ a.b }) = obj` vs `({ a.b } = obj)`?
